lib/ansible/modules/cloud/google/gcp_healthcheck.py

```diff
@@ -0,0 +1,447 @@
+  - Only supports HTTP and HTTPS Healthchecks currently.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
+  check_interval:
+    description:
+       - How often (in seconds) to send a health check. The default
+         value is 5 seconds.
+    required: false
+  healthcheck_name:
+    description:
+       - Name of the Healthcheck.
+    required: true
+  healthcheck_type:
+    description:
+       - Type of Healthcheck. HTTP or HTTPS.
+    required: true
+  host_header:
+    description:
+       - The value of the host header in the health check request. If left
+         empty (default value), the public IP on behalf of which this health
+         check is performed will be used.
+    required: true
+  port:
+    description:
+       - The TCP port number for health check request.The default value is 443
+         for HTTPS and 80 for HTTP.
+    required: false
+  request_path:
+    description:
+       - The request path of the HTTPS health check request.
+         The default value is "/".
+    required: false
+  timeout:
+    description:
+       - How long (in seconds) to wait before claiming failure. The default
+         value is 5 seconds. It is invalid for timeout to have a greater
+         value than check_interval.
+    required: false
+  unhealthy_threshold:
+    description:
+       - A so-far healthy instance will be marked unhealthy after this
+         many consecutive failures. The default value is 2.
+    required: false
+  healthy_threshold:
+    description:
+       - A so-far unhealthy instance will be marked healthy after this
+         many consecutive successes. The default value is 2.
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Create Minimum HealthCheck
+  gcp_healthcheck:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    healthcheck_name: my-healthcheck
+    healthcheck_type: HTTP
+    state: present
+- name: Create HTTP HealthCheck
+  gcp_healthcheck:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    healthcheck_name: my-healthcheck
+    healthcheck_type: HTTP
+    host: my-host
+    request_path: /hc
+    check_interval: 10
+    timeout: 30
+    unhealthy_threshhold: 2
+    healthy_threshhold: 1
+    state: present
+- name: Create HTTPS HealthCheck
+  gcp_healthcheck:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    healthcheck_name: "{{ https_healthcheck }}"
+    healthcheck_type: HTTPS
+    host_header: my-host
+    request_path: /hc
+    check_interval: 5
+    timeout: 5
+    unhealthy_threshold: 2
+    healthy_threshold: 1
+    state: present
+'''
+
+RETURN = '''
+state:
+    description: state of the Healthcheck
+    returned: Always.
+    type: str
+    sample: present
+updated_healthcheck:
```

Comment:
This isn't really needed - the task should already be returning changed==[true | false], so users interested in whether the resource was altered can just check that property instead.